### PR TITLE
feat: support `getLoadContext`

### LINKED ISSUE #5: Allow users to supply their own getLoadContext
According to [the Remix docs](https://remix.run/docs/en/main/guides/vite#augmenting-load-context):

> If you'd like to add additional properties to the load context, you should export a getLoadContext function from a shared module so that load context in Vite, Wrangler, and Cloudflare Pages are all augmented in the same way:

However, 

```ts
import handle from 'hono-remix-adapter/cloudflare-pages';
```

calls Remix with its own handler; not allowing the user to supply one

https://github.com/yusukebe/hono-remix-adapter/blob/4ad68d024236094042c614e485b913a541366836/src/cloudflare-pages.ts#L18-L28

This is a key bit of functionality. It'd be great if `handle` accepted another argument, or some other change was made so a user could supply their own `getLoadContext`

This is the expected type for the function

```ts
type GetLoadContext = (args: {
  request: Request;
  context: { cloudflare: Cloudflare }; // load context _before_ augmentation
}) => AppLoadContext;
```

### PR DESCRIPTION
**Pull request by yusukebe**
This PR enables users can access the Cloudflare values - `cf`, `ctx`, and `caches` on Remix routes.

Closes #5

### CODE REVIEW

**Comment by yusukebe** [comment]
Hey @jfsiii !

I've tried to intervene regarding the issue. Can you review it?

**Comment by jfsiii** [comment]
@yusukebe Adding those items to the context is a step in the right direction, but it still doesn't allow users to _add their own values_ to the context

As I quote in #5 , according to [the Remix docs](https://remix.run/docs/en/main/guides/vite#augmenting-load-context):

> If you'd like to add additional properties to the load context, you should export a `getLoadContext` function from a shared module so that load context in Vite, Wrangler, and Cloudflare Pages are all augmented in the same way:

The official handler from Remix also accepts the `getLoadContext`

```tsx
export const onRequest = createPagesFunctionHandler({
  build,
  getLoadContext,
});
```

Are you opposed to allowing this?  It's a critical feature for me. I have this patched/working locally and am happy to send a PR if that helps.

**Comment by yusukebe** [comment]
@jfsiii Thank you for the comment.

> Are you opposed to allowing this? It's a critical feature for me. I have this patched/working locally and am happy to send a PR if that helps.

No! It will be a good feature, and I'd like you to send a PR. So, first, I'll merge this PR, and after that, you will create a PR, okay?

**Comment by yusukebe** [comment]
I'm working on supporting `getLoadContext`. Please review it later.

**Comment by yusukebe** [comment]
@jfsiii 

I've added the code to support `getLoadContext` with user values. 

For dev server, you can add `getLoadContext` in `vite.config.ts`: https://github.com/yusukebe/hono-remix-adapter/pull/6/files#diff-6671dd857b3a02ae74576b82e0e7088e90b9c2220259fe16cf78a5833a2c6e10R24

For Cloudflare Pages, add it here: https://github.com/yusukebe/hono-remix-adapter/pull/6/files#diff-26d9f92c12cdb12bbf15be7f110c94f8bd7557c13b4929cb240829e6d4d13f68R7

What do you think about this?

File: src/remix.ts
Comment (jfsiii) [inline src/remix.ts:9]: It seems like we could do more than `any` here.

It might be tricky/impractical to use the "real" `Env` and `PlatformProxy` types, but I think this works

```suggestion
    cloudflare: {
      env: Record<string, unknown>
      cf: IncomingRequestCfProperties
      ctx: ExecutionContext
      caches: typeof caches
    }
```

File: src/remix.ts
Comment (jfsiii) [inline src/remix.ts:14]: Names are hard, but if you gave that type a name like:

```ts
type SomeContext = {
  cloudflare: {
    env: Record<string, unknown>
    cf: IncomingRequestCfProperties
    ctx: ExecutionContext
    caches: typeof caches
  }
};
```

We could use it like

```ts
export type GetLoadContext = (args: {
  request: Request
  context: SomeContext
}) => AppLoadContext

export const defaultGetLoadContext = ({ context }: { context: SomeContext}) => {
```

File: src/remix.ts
Comment (jfsiii) [inline src/remix.ts:24]: I wasn't sure about this one. Do we want `c.env` which will return

```
{
      "MY_VAR": "My Valuez",
      "ASSETS": {},
      "eventContext": {
        "request": {},
        "functionPath": "/",
        "params": {},
        "data": {},
        "env": {
          "MY_VAR": "My Valuez",
          "ASSETS": {}
        }
      }
}
```

or `c.env.eventContext.env`, which would only return

```
        {
          "MY_VAR": "My Valuez",
          "ASSETS": {}
        }
```

File: example/vite.config.ts
Comment (jfsiii) [reply example/vite.config.ts:24]: Also, I'm not 💯 in my testing, but when I added

```
    remixCloudflareDevProxy({
      getLoadContext,
    }),
```
and removed `getLoadContext` from `serverAdapter`, the example app still passed my bindings and augmented context through

**Review by jfsiii — APPROVED** [review]
**TL;DR** It works! Ship it 🚀 

I left lots of comments/questions, but nothing that I think should hold up the PR. They'd all be easy enough to follow up on later. Especially during this 0.x period

File: example/vite.config.ts
Comment (yusukebe) [reply example/vite.config.ts:24]: About `remixCloudflareDevProxy`, perhaps it may not need it because this `hono-remix-adapter` does the same thing. I'll test it; if so, I will create the PR.

File: src/cloudflare-pages.ts
Comment (yusukebe) [reply src/cloudflare-pages.ts]: Good suggestion. I've updated the code.

File: src/cloudflare-pages.ts
Comment (yusukebe) [reply src/cloudflare-pages.ts]: Updated.

File: src/remix.ts
Comment (yusukebe) [reply src/remix.ts:24]: This function is also used in the dev server: https://github.com/yusukebe/hono-remix-adapter/pull/6/files#diff-34d76a96faa930c4b9909c055beb422e601e2267f72ce12d6f15a1c2cc6ea914R23

It should be `c.env` to get `MY_VAR` correctly since it does not have `eventContext`.

File: src/remix.ts
Comment (yusukebe) [reply src/remix.ts:14]: If you override the definition of `AppLoadContext` in your app: https://github.com/yusukebe/hono-remix-adapter/pull/6/files#diff-fe3b5413a3ab56944fc97a3eb9b71e83c41e4d0b41a2303932ae92bd8c82cc8fR11-R13

The type mismatch will occur. To remove the typo error, I've added the `any`. This may be fixed in the future, but it remains as `any`.

Comment (jfsiii) [reply src/remix.ts:14]: @yusukebe Oh, right. Apologies for not testing that locally. I think 0.2.0 shipped with this type, because I'm getting this type error in my app after updating

Nothing a `@ts-expect-error` can't fix, though

Screenshot 2024-09-23 at 7 46 38 AM

Screenshot 2024-09-23 at 7 40 38 AM

Comment (yusukebe) [reply src/remix.ts:14]: @jfsiii 

Oooops. Sorry. The building way was wrong. Fixed it and released the new version. Try `v0.2.1`.

Comment (jfsiii) [reply src/remix.ts:14]: No worries. It happens. And it was kind of my fault, anyway.

Thanks!

**Pull request merged**
